qos/evm/observe.go

```diff
@@ -0,0 +1,69 @@
+// evm package provides the support required for interacting
+// with an EVM blockchain through the gateway.
+package evm
+
+import (
+	"context"
+)
+
+// observationSet provides all the functionality required
+// by the message package's ObservationSet to handle the sharing
+// of QoS data between PATH instances, and the updating of local
+// PATH instance's QoS data on EVM endpoints.
+var _ message.ObservationSet = observationSet{}
+
+// observation captures the result of processing an endpoint's
+// response to a service request.
+// It provides details needed to establish the validity of
+// an endpoint.
+type observation struct {
+	// TODO_IMPROVE: use a custom type here.
+	ChainID     string
+	BlockHeight uint64
+}
+
+type observationSet struct {
+	// TODO_IMPROVE: use an interface here.
+	EndpointStore *EndpointStore
+
+	Observations map[relayer.EndpointAddr][]observation
+}
+
+// TODO_IN_THIS_COMMIT: implement marshalling
+func (os observationSet) MarshalJSON() ([]byte, error) {
+	return nil, nil
+}
+
+func (os observationSet) NotifyStakeHolders() error {
+	if os.EndpointStore == nil {
+		return errors.New("notifyStakeHolders: endpoint store not set")
+	}
+
+	return os.EndpointStore.ApplyObservations(os.Observations)
+}
+
+// TODO_IMPROVE: use a separate function/struct here, instead of splitting
```

Comment:
Can we call this `ProcessObservations` rather than `Apply`?

Not a strong opinion.